packages/itwinui-css/src/searchbox/searchbox.scss

```diff
@@ -17,6 +17,7 @@
 .iui-expandable-searchbox {
   inline-size: var(--_iui-input-min-height);
   transition: width var(--iui-duration-1) ease-in-out;
+  box-sizing: content-box; // Center the searchbox-open-button (in the non-expanded state) within the expandable-searchbox
```

Comment:
🤔 Shouldn't we adjust the padding instead? Changing `box-sizing` feels like a hack

Reply:
I don't think that using `box-sizing` to center an element is the best solution either.

Reply:
Ohh, I see. The reason I was doing this was the fix the below issue in the [SearchBox → Expandable](https://itwin.github.io/iTwinUI/react/?story=searchbox--expandable&theme=dark) when we added `1px` padding-inline on `iui-input-flex-container`:

|Currently hosted version|This PR without `box-sizing: content-box`|This PR with `box-sizing: content-box`|
|-|-|-|
|<img width="231" alt="image" src="https://github.com/iTwin/iTwinUI/assets/45748283/82b8dbda-1b94-4cce-8936-99308e94a7fe">|<img width="233" alt="image" src="https://github.com/iTwin/iTwinUI/assets/45748283/8c2e3ab7-c1ff-4301-ac19-0beb2985b56a">|<img width="232" alt="image" src="https://github.com/iTwin/iTwinUI/assets/45748283/162c4e62-1c27-4390-877b-7301a7f98425">|
|Icon is centered|Icon is right of center|Icon is centered|

The `box-sizing: content-box` was to avoid including the padding into the size. With this, the new size is `38x36` (38 in width since it includes the 36px in content size + 2px in total inline-padding) and thus the icon/button is now centered.

I will look for alternatives to doing this. Open to suggestions.

<!--
|<img width="122" alt="image" src="https://github.com/iTwin/iTwinUI/assets/45748283/99313569-ef81-49ba-9fb6-190842aa40fc">|<img width="128" alt="image" src="https://github.com/iTwin/iTwinUI/assets/45748283/4d760aa9-258c-4884-aa42-9cafe72923a4">|
-->

---

> 🤔 Shouldn't we adjust the padding instead? Changing `box-sizing` feels like a hack

The 1px inline-padding is coming from iui-input-flex-container as a result of this PR. Are you suggesting setting it to 0px when iui-input-flex-container is used in iui-expandable-searchbox?

Reply:
Yes, we can remove the 1px padding in collapsed state

Reply:
As of now, I removed the 1px padding on .iui-expandable-searchbox regardless of the collapsed/expanded state ([commit](https://github.com/iTwin/iTwinUI/pull/1828/commits/ff19c1c28c4e474aa15d0a14ff5bf30a7c6ffa2b)).

This is because just removing it in the collapsed state is causing an abrupt shift of the search icon when going between the collapsed and expanded states:

https://github.com/iTwin/iTwinUI/assets/45748283/76a6770f-88ba-49c8-bc81-c2e0f07264a1

Reply:
Could you maybe use `padding-inline-end` then? It solves the main issue, avoids the layout shift, and simplifies the code.

Reply:
Yes, just padding-inline-end would solve the main issue and avoid the layout shift (https://github.com/iTwin/iTwinUI/pull/1828/commits/fe605e2cdd6ce2b36d47d6a46a6ace9fb154722a).

But if we want symmetry of the inline-paddings, we can also add a padding-inline-start on all iui-input-flex-containers. Then in the searchbox styles, we can remove padding-inline-start and set padding-inline-end only in the expanded case to prevent the layout shift and fix the searchbox-open-button centering issue. (https://github.com/iTwin/iTwinUI/pull/1828/files/fe605e2cdd6ce2b36d47d6a46a6ace9fb154722a..d90639418b162fc5c6807d2803cf7adaf0eceef7)

Reply:
That's fine with me. Can you also add a short comment next to `padding-inline-start: 0`, since it's not obvious why it's there.

Reply:
Added a comment to explain `padding-inline-start: 0` (https://github.com/iTwin/iTwinUI/pull/1828/commits/62f5e5747ea6e4977d341174f6357aa2a69bb18c).

This commit also adds a small comment for `padding-inline: 0`. But I can remove it if its reason is obvious without the comment.